app/models/queued_post.rb

```diff
@@ -73,10 +78,22 @@ def approve!(approved_by)
       unless created_post && creator.errors.blank?
         raise StandardError.new(creator.errors.full_messages.join(" "))
       end
+
+      if revised?
+        revisor = PostRevisor.new(created_post)
+        revisor.revise!(
+          User.find(post_options['changes']['editor_id']),
+          post_options['changes'],
+          skip_validations: true,
+          skip_jobs: true,
```

Comment:
My concern here is that you are introducing a timing problem 

If post is created, queues a post processor, then is revised ... no post processor runs, you will have a situation where cooked may not be correct. The post process is immediately queued from the post creator. 

In fact I would not do any of the transactioning here, or job omissions can happen, just create and revise outside of transaction, its simplest and if disaster hits you can add a cleanup the just deleted the post / topic.

Reply:
Could you please further explain the potential timing problem?

AFAICT, `PostRevisor` does not depend on `cooked`, and the post processor will be queued twice (once by `PostCreator` and another by `PostRevisor`) when the transaction block commits.

Removing the transaction block can be tricky, since it does quite a few things at the same time: change the status of itself (queued post), unblock the user, and create/revise the post. 

Will look into other possible solutions in the meantime.